File: julia/perfect-numbers/perfect-numbers.jl
function isperfect(number::Int)
    if number <= 0
        throw(DomainError(number, "Only Natural Numbers!"))
    end
    factors = [i for i in 1:number÷2 if number % i == 0]
    return sum(factors) == number
end

function isabundant(number::Int)
    if number <= 0
        throw(DomainError(number, "Only Natural Numbers!"))
    end
    factors = [i for i in 1:number÷2 if number % i == 0]
    return sum(factors) > number
end

function isdeficient(number::Int)
    if number <= 0
        throw(DomainError(number, "Only Natural Numbers!"))
    end
    factors = [i for i in 1:number÷2 if number % i == 0]
    return sum(factors) < number
end

File: python/perfect-numbers/perfect_numbers.py
def classify(number):
    if (number < 1):
        raise ValueError("Only natural numbers are allowed.")
    else:
        aliquot_sum = 0

        for i in range(1, int(number/2)+1):
            if number%i == 0:
                aliquot_sum += i
    
        if aliquot_sum == number:
            return "perfect"
        elif aliquot_sum > number:
            return "abundant"
        else:
            return "deficient"
